validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -21,51 +24,85 @@
 from sawtooth_validator.journal.consensus.consensus\
     import ForkResolverInterface
 
+from sawtooth_validator.state.config_view import ConfigView
+
 
 class BlockPublisher(BlockPublisherInterface):
-    """ DevMode BlockPublisher implementation.
+    """DevMode consensus uses genesis utility to configure Min/MaxWaitTime
+     to determine when to claim a block.
+     Default MinWaitTime to zero and MaxWaitTime is 0 or unset,
+     ValidBlockPublishers default to None or an empty list.
+     DevMode Consensus (BlockPublisher) will read these settings
+     from the StateView when Constructed.
     """
+
     def __init__(self, block_cache, state_view, batch_publisher, data_dir):
         super().__init__(block_cache, state_view, batch_publisher, data_dir)
 
         self._block_cache = block_cache
         self._state_view = state_view
-        self.batch_publisher = batch_publisher
-        self._num_batches = 1
-        self._count = 0
+
+        self._start_time = 0
+        self._wait_time = 0
+        config_view = ConfigView(self._state_view)
+        self._min_wait_time = config_view.get_setting(
+            "sawtooth.consensus.min_wait_time", 0, int)
+        self._max_wait_time = config_view.get_setting(
+            "sawtooth.consensus.max_wait_time", 0, int)
+        self._valid_block_publishers = config_view.get_setting(
+            "sawtooth.consensus.valid_block_publishers", None, list)
 
     def initialize_block(self, block_header):
-        """Initialize the candidate block_header.
+        """Do initialization necessary for the consensus to claim a block,
+        this may include initiating voting activates, starting proof of work
+        hash generation, or create a PoET wait timer.
+
         Args:
-            block_header (block_header): the block_header to initialize.
+            block_header (BlockHeader): the BlockHeader to initialize.
         Returns:
-            Boolean: True if the candidate block_header should be built.
+            True
         """
+        block_header.consensus = b"Devmode"
+        self._start_time = time.time()
+        self._wait_time = random.uniform(
+            self._min_wait_time, self._max_wait_time)
         return True
 
-    def check_publish_block(self, block_header):
-        """Check if a candidate block_header is ready to be claimed.
+    def check_publish_block(self, block):
```

Comment:
hmm not sure what happened when i rebased, but i did changed the argument to Block_header on Friday. I just changed it again.